[Docs] 공통 응답 설정 PR

rlqor9688 opened the pull request:
## 📌 관련 이슈
- Closes #11

## ✨ 변경 사항
- 1. ApiResponse : CommonResponseDto.java 클래스 수정
- 2. BaseCode : ReasonDto를 반환하는 메서드를 가진 인터페이스
- 3. ReasonDto : HttpStatus, isSuccess, message를 필드로 가지는 클래스
- 4. SuccessCode : BaseCode를 상속받아 HttpStatus, message를 필드로 가지는 Enum 클래스
- 5. ErrorCode : BaseErrorCode를 상속 받아 HttpStatus, message를 필드로 갖는 Enum 클래스
- 6. CustomException : BaseErrorCode를 필드로 갖는 클래스(RuntimeException 상속)
- 7. GlobalExceptionHandler : ApiResponse의 ErrorReasonDto를 반환하는 글로벌 익셉션 핸들러

(참고) Error/Code는 기존과 동일하게 Status와 message만 포함.

## 📷 Postman 
- 이미지 첨부

## ✅ 체크리스트
- [ ] 관련 이슈에 연결됨
- [ ] 기능이 정상적으로 동작함
- [ ] 코드 리뷰를 받았음
- [ ] 불필요한 로그/코드 제거됨

## 💬 기타 참고 사항
- 테스트 방법, 추가 설명 등 필요한 경우 작성해 주세요.

rlqor9688 committed:
[feat] Goal Entity 설계 임시 저장
<Entity>
1. Goal : 목표 Entity
2. GoalHousing : 주거 목표 Entity
3. GoalRetirement : 은퇴 목표 Entity
4. GoalEtc : 기타 목표 Entity
<Enum>
1. Category : 세부 목표 카테고리
2. HousingType : 주거 형태
3. Share : 공유 설정
4. Status : 목표 상태

rlqor9688 committed:
[docs] 공통 응답 설정
1. ApiResponse : CommonResponseDto.java 클래스 수정
2. BaseCode : ReasonDto를 반환하는 메서드를 가진 인터페이스
3. ReasonDto : HttpStatus, isSuccess, message를 필드로 가지는 클래스
4. SuccessCode : BaseCode를 상속받아 HttpStatus, message를 필드로 가지는 Enum 클래스
5. ErrorCode : BaseErrorCode를 상속 받아 HttpStatus, message를 필드로 갖는 Enum 클래스
6. CustomException : BaseErrorCode를 필드로 갖는 클래스(RuntimeException 상속)
7. GlobalExceptionHandler : ApiResponse의 ErrorReasonDto를 반환하는 글로벌 익셉션 핸들러

woneyee commented on src/main/java/org/example/lifechart/domain/goal/entity/Goal.java:
@NoArgsConstructor(access = AccessLevel.PROTECTED) 고려해주신다면 좋을 것 같습니다 외부 클래스에서 실수로 new 를 사용해 객체 생성을 못하도록 미리 접근 제한을 두는 방식입니다!!

rlqor9688 replied:
피드백 감사합니다! 테스트할 때 일부 필드만 설정해 목 데이터를 만드는 경우는 어떡하지? 생각해 물어보니 builder를 쓰면 이 문제도 해결할 수 있다고 하네요! 참고 부탁드려요! 말씀주신 설정 업데이트 했습니다! @AllArgsConstructor도 순서를 기억해야하는 생성자보다 빌더 패턴을 사용하도록 유도하는 목적으로 PRIVATE 설정을 걸어 에서 빌더 패턴을 많이 쓴다고 하네요! 이 설정도 추가해 머지 하겠습니다!

@NoArgsConstructor(access = AccessLevel.PROTECTED) 관련 설명
![image](https://github.com/user-attachments/assets/3de19cac-3854-440d-8656-be8bba458ef3)
![image](https://github.com/user-attachments/assets/c9677788-054d-44c9-94e0-037bc89ff6bc)

@AllArgsConstructor(access = AccessLevel.PRIVATE)
![image](https://github.com/user-attachments/assets/37a0698d-4794-4bd7-8cdb-0f71a78430cc)
![image](https://github.com/user-attachments/assets/50b35d2d-83a2-4b30-8fc3-76266bab9a29)

덕분에 하나 배웠습니다~ 감사합니다!

rlqor9688 commented:
앗 Goal Entity 임시저장 Commit&Push했던 것도 올라갔네요.. Goal쪽은 Goal 생성 메서드 올릴 때 업데이트될 수 있음을 감안해주시면 감사하겠습니다!

woneyee reviewed (approved):
goal엔티티 필요했는데 올려주셨네요!! 리뷰 하나 남겼는데 확인해주시면 감사드리겠습니다!!

byeongtaek12 commented on src/main/java/org/example/lifechart/common/exception/GlobalExceptionHandler.java:
@ControllerAdvice 라고 하신 이유가 있을까요?

rlqor9688 replied:
오 감사합니다 병택님! 저도 가져다 쓰면서 의아했는데 빨리 해야한다는 생각에 미처 찾아보지 못하고 그대로 적용했습니다. 찾아보니 RestControllerAdvice가 RestAPI 기반 백엔드 프로젝트에 더 적합하다고 하네요! 수정해서 머지하겠습니다! 아래 GPT한테 물어본 답변 공유드리니 참고 부탁드려요~

![image](https://github.com/user-attachments/assets/349dbadb-1f34-4fd0-ba93-3895cbe53e68)

byeongtaek12 reviewed (approved):
@ControllerAdvice로 하신 이유가 궁금한 것 말곤 딱히 없습니다. 수고하셨습니다.

rlqor9688 committed:
[Docs] GlobalExceptionHandler 및 Entity Annotation 수정
1. GlobalExceptionHandler : 기존 ControllerAdvice -> RestControllerAdvice로 annotation 수정
2. Goal, GoalHousing, GoalEtc, GoalRetirement 내 NoArgsConstructor(access = AccessLevel.PROTECTED) 패턴 추가

rlqor9688 committed:
[docs] Goal Entity Annotation 설정 수정
1. Goal, GoalHousing, GoalRetirement, GoalEtc @NoArgsConstructor(access = AccessLevel.PROTECTED), @AllArgsConstructor(access = AccessLevel.PRIVATE) 설정 추가
